Docs: Document new value mappings (#35067)
grafana/grafana: docs/sources/panels/value-mappings.md

weight = 600
+++

# Value mappings

Value mapping concept allow you to replace values or ranges in your visualizations with words or emojis.

Values mapped via value mappings will skip the unit formatting. This means that a text value mapped to a numerical value will not be formatted using the configured unit.

![Value mappings example](/static/img/docs/value-mappings/value-mappings-example-8-0.png)

If value mappings are present in a panel, then Grafana displays a summary in the side pane of the panel editor.

## Value mapping examples

Value mappings are displayed differently in different visualizations.

### Time series example

> 

Reviewer: Will be nice to have an introductory/stem sentence.

Author: Added

![Value mappings time series example](/static/img/docs/value-mappings/value-mappings-summary-example-8-0.png)

### Stat example

![Value mappings stat example](/static/img/docs/value-mappings/value-mappings-stat-example-8-0.png)

### Bar gauge example

![Value mappings bar gauge example](/static/img/docs/value-mappings/value-mappings-bar-gauge-example-8-0.png)

### Table example

![Value mappings table example](/static/img/docs/value-mappings/value-mappings-table-example-8-0.png)

## Value mappings fields

This section describes options in the Value mappings user interface.

> > **Note:** Display text and Color are both optional. If you only want to assign colors to text values, then you can leave the display text empty and the original value will be displayed.

Reviewer: ```suggestion
> **Note:** Display text and Color are both optional. If you want to only assign colors to the text values, then you can leave the display text empty and the original value is displayed.
```

weight = 600
+++

# Value mappings

Value mapping concept allow you to replace values or ranges in your visualizations with words or emojis.

Values mapped via value mappings will skip the unit formatting. This means that a text value mapped to a numerical value will not be formatted using the configured unit.

![Value mappings example](/static/img/docs/value-mappings/value-mappings-example-8-0.png)

If value mappings are present in a panel, then Grafana displays a summary in the side pane of the panel editor.

## Value mapping examples

Value mappings are displayed differently in different visualizations.

### Time series example

Here's an example showing a Time series visualization with value mappings.

![Value mappings time series example](/static/img/docs/value-mappings/value-mappings-summary-example-8-0.png)

> 

Reviewer 2: This time series example can be removed as the value mappings are not used in the time series panel. Here only the select color scheme (classic palette) is used to define series colors, and the series color is used for the line/area/points.  The value mappings only apply for visualizations that render individual values (with different colors).

Author: Except that the value mappings appeared on the axis. Is that not intentional?

Reviewer 2: Yes, for axis values. But no color. So if you want to show case axis values that is then don't set color as that is not used at all. See example in what's new in v7.4 dashboard on play, it has a value mapping example with a graph that show cases value mapping for y axis

Author: Okay. What I was trying to do with the examples was keep the same settings and just change the visualization, showing what the same settings would do for each.

Author: BTW, sometimes the value mappings disappear when you switch visualizations. I haven't found a pattern to it yet, just suddenly have to redo them all. Super annoying.

Reviewer 2: could be that some visualization do not support it (old graph, heatmap, text panel etc)

Reviewer 2: This one I would use for value time series
![Screenshot from 2021-06-02 19-35-03](https://user-images.githubusercontent.com/10999/120526615-ad6f3c80-c3d9-11eb-8d29-9002b6369d81.png)

Author: Pretty! I will add that in, thanks!

Author: Added in https://github.com/grafana/website/pull/4630

### Stat example

Here's an example showing a Stat visualization with value mappings.

> ![Value mappings stat example](/static/img/docs/value-mappings/value-mappings-stat-example-8-0.png)

Author: Thank you!

### Bar gauge example

Here's an example showing a Bar gauge visualization with value mappings.

![Value mappings bar gauge example](/static/img/docs/value-mappings/value-mappings-bar-gauge-example-8-0.png)

### Table example

Here's an example showing a Table visualization with value mappings.

> ![Value mappings table example](/static/img/docs/value-mappings/value-mappings-table-example-8-0.png)

Author: I could not get the colors to show when I experimented with it. Is there a special setting?

This column lists the type of condition a value mapping is triggered by and the values.

You can map values to three different conditions:

- **Value** maps text values to a color or different display text. For example, if a value is `10`, I want Grafana to display **Perfection!** rather than the number.
- **Range** maps numerical ranges to a display text and color. For example, if a value is within a certain range, I want Grafana to display **Low** or **High** rather than the number.
- **Special** maps special values like `Null`, `NaN` (not a number), and boolean values like `true` and `false` to a display text and color. For example, if Grafana encounters a `null`, I want Grafana to display **N/A**.

You can also use the dots on the left as a "handle" to drag and reorder value mappings in the list.

### Display text

The _display text_ is what Grafana displays instead of a number when the listed condition is met.

You can enter any Ascii character or emoji in this field.

### Color

You can select a color to for Grafana to display the value mapping text in.

- **Set color -** Click **Set color** to see a range of recommended colors. Click **Custom** to choose your own color.

> - **Text color -** Select this option to make the displayed text the selected color.

Reviewer 2: ```suggestion
- **Text color -** The primary text color for the current theme, i.e. white in dark theme and black in light theme. 
```

> - **Transparent -** TORKEL, what does this do?

Reviewer 2: > - **Text color -** Select this option to make the displayed text the selected color.

This is not correct, the color option does not necessarily affect the display text color, but backgrounds (depends on visualization used). It can affect display text (for example in stat panels, or table with cell display mode set `Color text`). This option sets the selected color be the curren't themes primary text color (ie white in dark theme black in light theme).

I would remove these, they are part of the color picker, not sure they should be part of the value mappings docs. Transparent is if you want a color fully transparent (Shows panel background), mainly useful in thresholds for graphs.

Author: Really? I got color changes, at least sometimes. 

Because they are in the UI, we should address them in the documentation. Can you suggest better definitions?
